.github/workflows/spec-update.yaml

```diff
@@ -0,0 +1,168 @@
+          BRANCH="spec-update/$CHOICE/$REF"
+          git fetch origin $BRANCH || true
+          git checkout -B $BRANCH origin/$BRANCH || git checkout -b $BRANCH
+          echo "branch=$BRANCH" >> "$GITHUB_OUTPUT"
+
+      - name: "Download specification file"
+        id: download
+        env:
+          GH_ENTERPRISE_TOKEN: ${{ secrets.GH_TOOLS_TOKEN }}
+        run: |
+          case $CHOICE in
+            core)
+              API_URL="$API_BASE_URL/cloudsdk/cloud-sdk-java-tests/contents/aicore.yaml?ref=$REF"
+              FILE_PATH='packages/ai-api/src/spec/AI_CORE_API.yaml'
+              ;;
+            grounding)
+              # TODO
+              exit 1
+              ;;
+            orchestration)
+              API_URL="$API_BASE_URL/AI/llm-orchestration/contents/src/spec/api.yaml?ref=$REF"
+              FILE_PATH='packages/orchestration/src/spec/api.yaml'
+              ;;
+          esac
+          
+          echo "Downloading $CHOICE specification file from $API_URL ..."
+          
+          gh api $API_URL -H "Accept: application/vnd.github.raw" > $FILE_PATH
+
+      - name: "Exit if there are no changes"
+        id: spec_diff
+        run: |
+          if [[ `git status --porcelain` ]]; then
+            echo "spec_diff=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "spec_diff=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - uses: sap/ai-sdk-js/.github/actions/setup@main
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        with:
+          node-version: 20
+
+      - name: "Generate"
+        id: generate
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        run: |
+          pnpm generate
+
+      - name: "Compile and Test"
+        id: compile
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        # Compilation can easily fail e.g. from re-namings and has to be fixed manually.
+        # Thus, this action raises the PR anyway and only reports success or failure of compilation and testing.
+        run: |
+          if pnpm compile ; then
+            echo "compilation_result=success" >> "$GITHUB_OUTPUT"
+            if pnpm test ; then
+                echo "test_result=success" >> "$GITHUB_OUTPUT"
+            else
+                echo "test_result=failure" >> "$GITHUB_OUTPUT"
+            fi
+          else
+            echo "compilation_result=failure" >> "$GITHUB_OUTPUT"
+            echo "test_result=skipped" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: "Push changes"
+        id: push
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        run: |
+          git config --global user.email "cloudsdk@sap.com"
+          git config --global user.name "SAP Cloud SDK Bot"
+          git add .
+          git commit -m "Update $CHOICE based on $REF"
+          git push --set-upstream origin ${{ steps.branch.outputs.branch }}
+
+      - name: "Create PR"
+        if: ${{ env.CREATE_PR == 'true' && steps.spec_diff.outputs.spec_diff == 'true'}}
```

Comment:
[q] Not entirely understanding the whole process. And this is rather a question related to the internal repo. Why is `--field create-pr=false` always set to false in the internal workflow?

Reply:
Discussed.